pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -476,44 +473,77 @@ func (c *Controller) addNodeRoute(nodeName string, node *corev1.Node) error {
 			return nil
 		}
 		peerGatewayIP := ip.NextIP(peerPodCIDRAddr)
-		peerConfig[peerPodCIDR] = peerGatewayIP
-		podCIDRs = append(podCIDRs, peerPodCIDR)
+		peerConfigs[peerPodCIDR] = peerGatewayIP
+		peerPodCIDRs = append(peerPodCIDRs, peerPodCIDR)
+		peerNodeIP := peerNodeIPs.IPv4
+		if peerGatewayIP.To4() == nil {
+			peerNodeIP = peerNodeIPs.IPv6
+		}
+
+		klog.InfoS("Adding route and flow to Node", "Node", nodeName, "podCIDR", podCIDR,
+			"peerNodeIP", peerNodeIP)
 	}
 
-	ipsecTunOFPort := int32(0)
+	var ipsecTunOFPort uint32
 	if c.networkConfig.EnableIPSecTunnel {
 		// Create a separate tunnel port for the Node, as OVS IPSec monitor needs to
 		// read PSK and remote IP from the Node's tunnel interface to create IPSec
 		// security policies.
-		if ipsecTunOFPort, err = c.createIPSecTunnelPort(nodeName, peerNodeIP); err != nil {
-			return err
+		createIPSecTunnelPort := func(ip net.IP) (uint32, error) {
+			port := int32(0)
+			if port, err = c.createIPSecTunnelPort(nodeName, ip); err != nil {
+				return 0, err
+			}
+			return uint32(port), nil
+		}
+		if peerNodeIPs.IPv4 != nil {
+			port, err := createIPSecTunnelPort(peerNodeIPs.IPv4)
+			if err != nil {
+				return err
+			}
+			ipsecTunOFPort = port
+		} else if peerNodeIPs.IPv6 != nil {
+			port, err := createIPSecTunnelPort(peerNodeIPs.IPv6)
+			if err != nil {
+				return err
+			}
+			ipsecTunOFPort = port
+		} else {
+			return fmt.Errorf("there is no valid IP in peerNodeIPs")
```

Comment:
The `createIPSecTunnelPort` doesn't seem necessary since it just wraps one call. The peerNodeIPs check is also unncessary as we have verified it must contains at least one IPv4 or IPv6.
The code can just be:
```
peerNodeIP := peerNodeIPs.IPv4
if peerNodeIP == nil {
	peerNodeIP = peerNodeIPs.IPv6
}
port, err := c.createIPSecTunnelPort(nodeName, peerNodeIP)
if err != nil {
	return err
}
ipsecTunOFPort = uint32(port)
```

Reply:
Done